test/e2e/telemetry_logs_analysis_test.go

```diff
@@ -148,80 +158,99 @@ var _ = Describe(suite.ID(), Label(suite.LabelTelemetryLogsAnalysis), Ordered, f
-		})
+			assert.DeploymentReady(ctx, k8sClient, types.NamespacedName{Namespace: namespace, Name: logBackendName})
+			assert.DeploymentReady(ctx, k8sClient, types.NamespacedName{Namespace: namespace, Name: metricBackendName})
+			assert.DeploymentReady(ctx, k8sClient, types.NamespacedName{Namespace: namespace, Name: traceBackendName})
 
-		It("Should have running pipelines", func() {
-			assert.LogPipelineHealthy(ctx, k8sClient, otelCollectorLogPipelineName)
-			assert.MetricPipelineHealthy(ctx, k8sClient, metricPipelineName)
-			assert.TracePipelineHealthy(ctx, k8sClient, tracePipelineName)
+			assert.DeploymentReady(ctx, k8sClient, types.NamespacedName{Namespace: namespace, Name: otelCollectorLogBackendName})
+			assert.DeploymentReady(ctx, k8sClient, types.NamespacedName{Namespace: namespace, Name: fluentBitLogBackendName})
+			assert.DeploymentReady(ctx, k8sClient, types.NamespacedName{Namespace: namespace, Name: selfMonitorLogBackendName})
 		})
 
-		It("Should have a running metric agent daemonset", func() {
+		It("Should have a running agents", func() {
 			assert.DaemonSetReady(ctx, k8sClient, kitkyma.MetricAgentName)
+			assert.DaemonSetReady(ctx, k8sClient, kitkyma.FluentBitDaemonSet)
+		})
+
+		It("Should have running pipelines", func() {
+			assert.LogPipelineHealthy(ctx, k8sClient, logBackendName)
+			assert.MetricPipelineHealthy(ctx, k8sClient, metricBackendName)
+			assert.TracePipelineHealthy(ctx, k8sClient, traceBackendName)
+
+			assert.LogPipelineHealthy(ctx, k8sClient, otelCollectorLogBackendName)
+			assert.LogPipelineHealthy(ctx, k8sClient, fluentBitLogBackendName)
+			assert.LogPipelineHealthy(ctx, k8sClient, selfMonitorLogBackendName)
 		})
 
 		It("Should push metrics successfully", func() {
-			assert.MetricsFromNamespaceDelivered(proxyClient, metricbackendExportURL, otelCollectorNs, telemetrygen.MetricNames)
+			assert.MetricsFromNamespaceDelivered(proxyClient, metricBackendURL, namespace, telemetrygen.MetricNames)
 		})
 
 		It("Should push traces successfully", func() {
-			assert.TracesFromNamespaceDelivered(proxyClient, tracebackendExportURL, otelCollectorNs)
+			assert.TracesFromNamespaceDelivered(proxyClient, traceBackendURL, namespace)
 		})
 
-		It("Should not have any ERROR/WARNING logs in the OtelCollector containers", func() {
+		It("Should collect logs successfully", func() {
+			assert.LogsDelivered(proxyClient, "", logBackendURL)
+		})
+
+		It("Should collect otel collector component logs successfully", func() {
+			assert.LogsDelivered(proxyClient, "telemetry-", otelCollectorLogBackendURL)
+		})
+
+		It("Should collect fluent-bit component logs successfully", func() {
+			assert.LogsDelivered(proxyClient, "telemetry-", fluentBitLogBackendURL)
+		})
+
+		It("Should collect self-monitor component logs successfully", func() {
+			assert.LogsDelivered(proxyClient, "telemetry-", selfMonitorLogBackendURL)
+		})
+
+		It("Should not have any error/warn logs in the otel collector component containers", func() {
```

Comment:
I like that the code looks simpler and more intuitive now that everything is in the same context. But this approach will make things harder for someone who wants to check exactly what failed, since if the otel It statement fails, for example, the other two (fluentbit, selfmon) will be skipped. Maybe we can discuss this during the daily.

Reply:
It is fine to have everything in the same context (as discussed in the daily).